docs/_posts/examples/3400-01-26-game-controls.html

```diff
@@ -0,0 +1,133 @@
+    <button class='unround button icon caret-down' id='down'></button>
+    <button class='round-right button icon caret-right' id='right'></button>
+</div>
+
+<script>
+    //Initialize the map
+    var map = new mapboxgl.Map({
+        container: 'map',
+        style: 'mapbox://styles/mapbox/dark-v9',
+        center: [-87.6298, 41.8781],
+        zoom: 17,
+        bearing: -12,
+        pitch: 60
+    });
+
+    var deltaDistance = 100; // how far the map moves in pixels when the up or down arrow is clicked
+    var deltaDegrees = 25; // rotation of the map in degrees when the left or right arrow is clicked
+
+    // prevent zooming and panning that is not done
+    // with the keyboard arrows or buttons
+    map.boxZoom.disable();
+    map.dragPan.disable();
+    map.doubleClickZoom.disable();
+    map.scrollZoom.disable();
+    map.keyboard.disable();
+    map.touchZoomRotate.disable();
+
+    function easeTo(t) {
+        return t * (2 - t);
+    }
+
+    function move(pos, bearing) {
+        if (bearing) {
+            map.easeTo({
+                bearing: pos,
+                easing: easeTo
+            });
+        } else {
+            map.panBy(pos, {
+                easing: easeTo
+            });
+        }
+    }
+
+    function goDirection(dir) {
+        if (dir === 'left') {
+            move(map.getBearing() - deltaDegrees, true);
+        } else if (dir === 'right') {
+            move(map.getBearing() + deltaDegrees, true);
+        } else if (dir === 'up') {
+            move([0, -deltaDistance]);
+        } else if (dir === 'down') {
+            move([0, deltaDistance]);
+        }
+    }
+
+    document.body.addEventListener('keydown', function(e) {
+        e.preventDefault();
+        if (e.which === 38) { // up
+            goDirection('up');
+        } else if (e.which === 40) { // down
+            goDirection('down');
+        } else if (e.which === 37) { // left
+            goDirection('left');
+        } else if (e.which === 39) { // right
+            goDirection('right');
+        }
+    }, true);
+
+
+    var buttonLeft = ['left', document.getElementById('left')];
+    var buttonRight = ['right', document.getElementById('right')];
+    var buttonTop = ['up', document.getElementById('up')];
+    var buttonBottom = ['down', document.getElementById('down')];
+
+    var buttons = [buttonLeft, buttonRight, buttonTop, buttonBottom];
+    var persist;
+
+    function buttonStart(b) {
+        persist = setInterval(function() {
+            goDirection(b[0]);
+        }, 20);
+    }
+
+    function buttonEnd() {
+        clearInterval(persist);
+    }
+
+    buttons.forEach(function(b) {
+        b[1].addEventListener('mousedown', buttonStart.bind(this, b));
+        b[1].addEventListener('touchstart', buttonStart.bind(this, b));
+        b[1].addEventListener('mouseup', buttonEnd.bind(this, b));
+        b[1].addEventListener('touchend', buttonEnd.bind(this, b));
+    });
```

Comment:
Do we really need buttons AND keyboard shortcuts? A lot of code could be eliminated & simplified if we dropped the buttons.